### DISCUSSION TITLE
How to set modal instead redirection when using custom provider

### DISCUSSION THREAD

**Discussion by hyunoosung** [nextauthjs/next-auth#4336] [Help]
Following  OAuth provider from Using a custom provider(https://next-auth.js.org/configuration/providers/oauth#using-a-custom-provider)

Everything works but I wonder if this can be a modal instead redirection.

I've searched issues and discussions page with 'popup', 'modal', 'custom provider' but got no luck so far.

Please help me

How to reproduce ☕️
Follow below code in [Using a custom provider](Using a custom provider)
```
{
  id: "google",
  name: "Google",
  type: "oauth",
  wellKnown: "https://accounts.google.com/.well-known/openid-configuration",
  authorization: { params: { scope: "openid email profile" } },
  idToken: true,
  checks: ["pkce", "state"],
  profile(profile) {
    return {
      id: profile.sub,
      name: profile.name,
      email: profile.email,
      image: profile.picture,
    }
  },
}
```

Contributing 🙌🏽
Yes, I am willing to help answer this question in a PR

**Comment by arye321** [comment]
working examples:

TS: https://github.com/krushilnaik/with-prisma-mongodb-nextauth

simple JS example: https://github.com/arye321/nextauth-google-popup-login

**Reply by jsjoeio** [reply]
Thank you for sharing these! The second one is super helpful - now I just have to convert it to TypeScript :D

**Reply by arye321** [reply]
first one is typescript

**Reply by jsjoeio** [reply]
True but it's much easier to look at the JS one and convert than dig through the TS one
